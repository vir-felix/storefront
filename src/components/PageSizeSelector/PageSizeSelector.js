import React, { Component } from "react";
import PropTypes from "prop-types";
import Select from "components/Select";
import { observer, inject } from "mobx-react";

const PAGE_SIZES = [
  {
    name: "20 Products",
    value: 20
  },
  {
    name: "60 Products",
    value: 60
  },
  {
    name: "100 Products",
    value: 100
  }
];

@inject("uiStore")
@observer
class PageSizeSelector extends Component {
  static propTypes = {
    uiStore: PropTypes.object.isRequired
  }

  handleChange = (event) => {
    this.props.uiStore.setPageSize(event.target.value);
  }

  render() {
    const { uiStore } = this.props;

    return (
      <Select
        value={uiStore.pageSize}
        options={PAGE_SIZES}
        inputProps={{
          name: "pageSize",
          id: "page-size"
        }}
        onChange={this.handleChange}
      />
    );
  }
}

export default PageSizeSelector;
