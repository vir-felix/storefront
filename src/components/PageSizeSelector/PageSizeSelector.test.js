import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "mobx-react";
import PageSizeSelector from "./PageSizeSelector";

const uiStore = {
  pageSize: 20,
  setPageSize: jest.fn()
};


test("basic snapshot", () => {
  const component = renderer.create((
    <Provider uiStore={uiStore}>
      <PageSizeSelector />
    </Provider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
