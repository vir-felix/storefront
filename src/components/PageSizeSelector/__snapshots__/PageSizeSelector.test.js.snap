// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiInput-root-6 MuiInput-underline-10"
>
  <div
    className="MuiSelect-root-1"
  >
    <div
      aria-haspopup="true"
      aria-owns={null}
      aria-pressed="false"
      className="MuiSelect-select-2 MuiSelect-selectMenu-3 MuiInput-input-14"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      20 Products
    </div>
    <input
      aria-invalid={undefined}
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      defaultValue={undefined}
      id="page-size"
      name="pageSize"
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="hidden"
      value={20}
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-19 MuiSelect-icon-5"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
